Mathlib/AlgebraicTopology/SimplicialSet/Path.lean

```diff
@@ -1,140 +1,261 @@
+      Fin.coe_castSucc, Fin.coe_eq_castSucc]
+  arrow_tgt i := by
+    dsimp only [tr, trunc, SimplicialObject.Truncated.trunc, incl,
+      whiskeringLeft_obj_obj, id_eq, Functor.comp_map, Functor.op_map,
+      Quiver.Hom.unop_op]
+    rw [← FunctorToTypes.map_comp_apply, ← op_comp, ← tr_comp, δ_zero_mkOfSucc]
+
+/-- Further truncating `X` above `m` does not change the `m`-spine. -/
+lemma trunc_spine (j m : ℕ)
+    (h : m ≤ j + 1 := by omega) (hn : j ≤ n := by omega) :
+    ((trunc (n + 1) (j + 1)).obj X).spine m = X.spine m := rfl
+
+lemma spine_map_vertex (m : ℕ) (hm : m ≤ n + 1) (Δ : X _[m]ₙ₊₁)
+    (a : ℕ) (ha : a ≤ n + 1) (φ : [a]ₙ₊₁ ⟶ [m]ₙ₊₁) (i : Fin (a + 1)) :
+    (X.spine a ha (X.map φ.op Δ)).vertex i =
+      (X.spine m hm Δ).vertex (φ.toOrderHom i) := by
+  dsimp only [spine_vertex]
+  rw [← FunctorToTypes.map_comp_apply, ← op_comp, ← tr_comp, const_comp]
+
+lemma spine_map_subinterval (m : ℕ) (h : m ≤ n + 1) (j l : ℕ) (hjl : j + l ≤ m)
+    (Δ : X _[m]ₙ₊₁) :
+    X.spine l (by omega) (X.map (tr (subinterval j l hjl)).op Δ) =
+      (X.spine m h Δ).interval j l hjl := by
+  ext i
+  · dsimp only [Path.interval, Path₁.interval, spine_vertex]
+    rw [← FunctorToTypes.map_comp_apply, ← op_comp, ← tr_comp,
+      const_subinterval_eq j l hjl]
+  · dsimp only [Path.interval, Path₁.interval, spine_arrow]
+    rw [← FunctorToTypes.map_comp_apply, ← op_comp, ← tr_comp,
+      mkOfSucc_subinterval_eq]
+
+end Truncated
+
+variable (X : SSet.{u})
+
+/-- A path of length `n` in a simplicial set `X` is defined by 1-truncating `X`,
+then taking the 1-truncated path. -/
+abbrev Path (n : ℕ) := truncation 1 |>.obj X |>.Path₁ n
+
+namespace Path
+open Truncated (Path₁)
+
+variable {X} {n : ℕ}
+
+/-- To show two paths equal it suffices to show that they have the same edges. -/
+@[ext]
+lemma ext' {f g : Path X (n + 1)} (h : ∀ i, f.arrow i = g.arrow i) : f = g :=
+  Path₁.ext' h
+
+/-- For `j + l ≤ n`, a path of length `n` restricts to a path of length `l`, namely
+the subpath spanned by the vertices `j ≤ i ≤ j + l` and edges `j ≤ i < j + l`. -/
+abbrev interval (f : Path X n) (j l : ℕ) (h : j + l ≤ n := by omega) : Path X l :=
+  Path₁.interval f j l h
+
+variable {X Y : SSet.{u}} {n : ℕ} (f : Path X n) (σ : X ⟶ Y)
+
+/-- Maps of `n + 1`-truncated simplicial sets induce maps of paths. -/
+abbrev map : Path Y n := Path₁.map f <| truncation 1 |>.map σ
+
+lemma map_vertex (i : Fin (n + 1)) :
+    (f.map σ).vertex i = σ.app (op [0]) (f.vertex i) := rfl
+
+lemma map_arrow (i : Fin n) :
+    (f.map σ).arrow i = σ.app (op [1]) (f.arrow i) := rfl
+
+/-- `Path.map` respects subintervals of paths. -/
+lemma map_interval (j l : ℕ) (h : j + l ≤ n) :
+    (f.map σ).interval j l h = (f.interval j l h).map σ := rfl
+
+end Path
+
+/-- The spine of an `n`-simplex in `X` is the path of edges of length `n` formed
+by traversing in order through the vertices of `X _[n]ₙ₊₁`. -/
+abbrev spine (n : ℕ) : X _[n] → Path X n :=
+  truncation (n + 1) |>.obj X |>.spine n
```

Comment:
Does it make sense that we would take the `n`-spine in a simplicial set `X` via the `n + 1`-truncation of `X`? Or perhaps the spine should just not be defined for `n = 0`?